test/type-scavenger/equivalence.ll

```diff
@@ -0,0 +1,79 @@
+; Test if llvm-spirv type scavenging has an assertion 
+; failure due to incorrect lookup of an equivalence class leader.
+
+; RUN: llvm-as < %s | llvm-spirv -o %t.spv
+; RUN: spirv-val %t.spv
+; RUN: llvm-spirv %t.spv -o - -to-text | FileCheck %s --check-prefix=CHECK-SPIRV
+; RUN: llvm-spirv %t.spv -o - -r | llvm-dis | FileCheck %s --check-prefix=CHECK-LLVM
+
+; Incorrect lookup of equivalence class leader caused an assertion failure when
+; processing call instruction to this name
+; CHECK-SPIRV: _ZN4sycl3_V16marrayIiLm5EEixEm
+; CHECK-LLVM: _ZN4sycl3_V16marrayIiLm5EEixEm
+
+target datalayout = "e-i64:64-v16:16-v24:32-v32:32-v48:64-v96:128-v192:256-v256:256-v512:512-v1024:1024-n8:16:32:64"
+target triple = "spir64-unknown-unknown"
+
+define spir_func void @_ZN4sycl3_V18lgamma_rINS0_6marrayINS0_6detail9half_impl4halfELm5EEENS0_9multi_ptrINS2_IiLm5EEELNS0_6access13address_spaceE0ELNS9_9decoratedE0EEEEENSt9enable_ifIXaaaasr6detailE11is_marray_vIT_Esr6detailE20is_valid_elem_type_vISE_fdS5_Eaaaasr6detailE14is_multi_ptr_vIT0_Esr6detailE26has_writeable_addr_space_vISF_Esr6detailE20is_valid_elem_type_vISF_NS3_15change_elementsIiSE_E4typeEEESE_E4typeESE_SF_() {
```

Comment:
You can simplify the function names here so they're not overly long Itanium-mangled names.

Reply:
Function names shortened